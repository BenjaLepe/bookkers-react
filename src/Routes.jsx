import React from 'react';
import { Route, Switch } from 'react-router-dom';
import NotFound from './views/NotFound';
import Login from './views/Login';
import BookDetail from './views/BookDetail';
import SignUp from './views/SignUp';
import Home from './views/Home';
import AuthContextProvider from './contexts/AuthContext';
import SessionProfile from './views/SessionProfile';
import SessionExpired from './views/SessionExpired';
import UserProfile from './views/UserProfile';
import CreateBook from './views/AddBook';
import EditUser from './views/EditUser';
import Navbar from './Components/Navbar';
import reviewReports from './views/Reports';
import EditBook from './views/EditBook';

export default function Routes() {
  return (
    <AuthContextProvider>
      <Navbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/sign-up" component={SignUp} />
        <Route exact path="/session-expired" component={SessionExpired} />
        <Route exact path="/session-profile" component={SessionProfile} />
        <Route exact path="/users/:id" component={UserProfile} />
        <Route exact path="/books/new" component={CreateBook} />
        <Route exact path="/books/:id" component={BookDetail} />
        <Route exact path="/edit-book/:id" component={EditBook} />
        <Route exact path="/edit-user/:id" component={EditUser} />
        <Route exact path="/books/:book_id/reviews/:review_id/reports" component={reviewReports} />
        <Route component={NotFound} />
      </Switch>
    </AuthContextProvider>
  );
}