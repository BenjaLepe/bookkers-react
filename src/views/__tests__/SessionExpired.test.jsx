import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../../Routes';

const { create } = renderer;

function TestRouter({ path }) {

  return (
    <MemoryRouter initialEntries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

describe('SessionExpires', () => {
  describe('When the session ends it', () => {
    it ('renders the session expired view with a home button', () => {
      const DOM = create(<TestRouter path ="/session-expired" />).toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });
});