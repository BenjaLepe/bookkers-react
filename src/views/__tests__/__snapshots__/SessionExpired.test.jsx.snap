// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionExpires When the session ends it renders the session expired view with a home button 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/sign-up"
          onClick={[Function]}
        >
          Sign Up 
        </a>
      </div>
    </div>
  </div>,
  <div>
    <h2>
      Session Expires
    </h2>
    <p>
      Your sessi√≥n has expired, please log in again :)
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        type="button"
      >
        Home
      </button>
    </a>
  </div>,
]
`;
